tensorflow/lite/micro/recording_micro_allocator.cc

```diff
@@ -18,12 +18,29 @@ limitations under the License.
 #include "tensorflow/lite/core/api/error_reporter.h"
 #include "tensorflow/lite/kernels/internal/compatibility.h"
 #include "tensorflow/lite/micro/compatibility.h"
+#include "tensorflow/lite/micro/memory_helpers.h"
 #include "tensorflow/lite/micro/memory_planner/greedy_memory_planner.h"
 #include "tensorflow/lite/micro/micro_allocator.h"
+#include "tensorflow/lite/micro/micro_error_reporter.h"
 #include "tensorflow/lite/micro/recording_simple_memory_allocator.h"
 
 namespace tflite {
 
+size_t RecordingMicroAllocator::GetDefaultTailUsage() {
+  // RecordingMicroAllocator inherits from MicroAllocator and its tail usage is
+  // similar with MicroAllocator with SimpleMemoryAllocator and MicroAllocator
+  // being replaced.
+  return MicroAllocator::GetDefaultTailUsage(
+             /*is_memory_planner_given=*/false) +
+         AlignSizeUp(sizeof(RecordingSimpleMemoryAllocator),
+                     alignof(RecordingSimpleMemoryAllocator)) -
```

Comment:
What do you think of adding a  templated AlignSizeUp

```cc
template <typename T>
size_t AlignSizeUp() {
  AlignSizeUp(sizeof(T), alignof(T));
}
```

That would make this expression easier to read:

```
 return AlignSizeUp<RecordingSimpleMemoryAllocator>() - AlignSizeUp<SimpleMemoryAllocator>() +
         AlignSizeUp<RecordingMicroAllocator>() -      AlignSizeUp<MicroAllocator>()

Reply:
Create a bug http://b/208703041 and add it into the TODO. Prefer another PR for it to keep the change scope smaller in this PR.